realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -303,4 +334,5 @@ static void simulateClientReset(SyncSession session) {
     protected static native void nativeInitializeSyncManager(String syncBaseDir);
     private static native void nativeReset();
     private static native void nativeSimulateSyncError(String realmPath, int errorCode, String errorMessage, boolean isFatal);
+    private static native void reconnect();
```

Comment:
@cmelchior 
I'm not sure about the Slack example.
 
I believe calling reconnect is a _blocking_ call in Slack kind of UI. Whereas `reconnect` here just reset the reconnect_delay, and send a `ping` to the server. The SyncClient may not resume Uploading directly.

I'm also afraid users will try to register another `NetworkListener` or JobScheduler to listen to network changes in order to call `reconnect`, which is redundant.

If something network related changed then our listener will be triggered (Slack maybe doing polling hence the force refresh now ...)

IMO. either we expose the method only and let the user, manually call (or register a NetworkListener to do it for him) *or* rely on our internal listener to call the method.